src/ctap/credential_id.rs

```diff
@@ -161,12 +149,12 @@ fn remove_padding(data: &mut Vec<u8>) -> Result<(), Ctap2StatusCode> {
     Ok(())
 }
 
-impl CredentialId {
+impl ParsedCredentialId {
     /// Encrypts the given private key, relying party ID hash, and cred protect policy into a credential ID.
     ///
     /// Other information, such as a user name, are not stored. Since encrypted credential IDs are
     /// stored server-side, this information is already available (unencrypted).
-    pub(super) fn encrypt_to_bytes(
+    pub(super) fn encrypt(
```

Comment:
Curious about the design choice here. `encrypt` is part of `impl`, but doesn't use `&self`, even though it takes a `&ParsedCredentialId`. Couldn't we call it like
`private_key.encrypt(env, rp_id_hash, cred_protect_policy)`
instead of
`ParsedCredentialId::encrypt(env, private_key, rp_id_hash, cred_protect_policy)`
?

Reply:
Private key is not "more important" than other fields it has, so I don't want it to be a method on private key.

More logically it should be `ParsedCredentialId::new(...).encrypt()`, but this requires ownership and cloning, but in fact only the reference of those fields are ever needed if we are only going to call encrypt() anyway. Any better pattern suggestion?

Reply:
Ah sorry, the `private_key` is not even a `ParsedCredentialId` anyway. So this function is unrelated to `ParsedCredentialId`. And it is also wrapped by `encrypt_key_handle` as a one-liner. Means we could rename it to `encrypt_key_handle` and then simply import it in `crypto_wrapper.rs`.

This also means that `ParsedCredentialId` is a private type used as an intermediate output of credential parsing, but ultimately converted to a CredentialSource. Maybe we can move pieces around to make everything in `ParsedCredentialId` private (or just delete it)?

Reply:
Made a quite big change by directly putting encrypt & decrypt inside credential_id.rs without exposing any other intermediate types anymore. And make some naming changes that is more correct.